src/test/java/gumtree/spoon/TreeTest.java

```diff
@@ -538,4 +538,32 @@ public void test_nestingLevelOfSuperType() {
 		assertEquals(superClass.getDepth(), klass.getDepth()+1);
 		assertEquals(0, superClass.getDescendants().size());
 	}
+
+	@Test
+	public void test_levelsOfNestingOfTypeArguments() throws Exception{
+		File nestedList = new File("src/test/resources/examples/diffOfGenericTypeReferences/multipleNesting/right.java");
+
+		AstComparator comparator = new AstComparator();
+		SpoonGumTreeBuilder scanner = new SpoonGumTreeBuilder();
+		ITree root = scanner.getTree(comparator.getCtType(nestedList));
+
+		String LIST_LABEL = "java.util.List";
+		String STRING_LABEL = "java.lang.String";
+		int listCounter = 0;
+		int stringCounter = 0;
+
+		ITree field = root.getChild(0).getChild(3);
+		for (ITree typeArgument: field.getDescendants()) {
+			if (typeArgument.getLabel().equals(LIST_LABEL)) {
+				++listCounter;
+			} else if (typeArgument.getLabel().equals(STRING_LABEL)) {
+				++stringCounter;
+			} else {
+				throw new Exception("Test case should only have a nested list of string");
+			}
+		}
+
+		assertEquals("Not enough List containers", 4, listCounter);
+		assertEquals("Not enough String type arguments", 1, stringCounter);
```

Comment:
@algomaster99 Regarding your question about writing this differently, I think what you've got here is fine, but I personally most often prefer using streams over loops. Here one could for example consider something like this:

```suggestion
		String listLabel = "java.util.List";
		String stringLabel = "java.lang.String";

		ITree field = root.getChild(0).getChild(3);
		var childLabels = field.getDescendants().stream().map(ITree::getLabel).collect(Collectors.toList());
		long numListLabels = childLabels.stream().filter(listLabel::equals).count();
		long numStringLabels = childLabels.stream().filter(stringLabel::equals).count();

		assertEquals("Not enough List containers", 4, numListLabels);
		assertEquals("Not enough String type arguments", 1, numStringLabels);
		assertEquals("There should only be list and string labels", childLabels.size(), numListLabels + numStringLabels);
```
I just think it's cleaner and easier to reason about.

On a side note, I would strongly advice against:

* Naming local variables `LIKE_THIS`, as that's a naming convention for static, final fields (constants)
* Throwing an exception in a test, prefer using the `fail()` method from JUnit instead

Reply:
> assertEquals("There should only be list and string labels", childLabels.size(), numListLabels + numStringLabels);

This is better than throwing an exception for sure.

Thanks!

Reply:
No problem!